File: JavaDemo/JavaWebDemo/src/main/java/com/example/javawebdemo/utils/DBUtils.java
package com.example.javawebdemo.utils;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DBUtils {
    private static Connection connection = null;

    public static Connection getConnection() {
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            final String url = "jdbc:mysql://127.0.0.1:3306/userinfo";
            final String username = "root";
            final String password = "123456";
            connection = DriverManager.getConnection(url, username, password);
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
        return connection;
    }

    public static void closeConnection() {
        if (connection != null) {
            try {
                connection.close();
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }
    }
}

File: KotlinDemo/JavaWebDemo/src/main/kotlin/com/example/javawebdemo/utils/DBUtils.kt
package com.example.javawebdemo.utils

import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

class DBUtils {
    companion object {
        val connection: Connection
            get() {
                Class.forName("com.mysql.cj.jdbc.Driver")
                val url = "jdbc:mysql://127.0.0.1:3306/userinfo"
                val username = "root"
                val password = "123456"
                return DriverManager.getConnection(url, username, password)
            }

        fun closeConnection(){
            try{
                connection.close()
            }catch (e:SQLException){
                e.printStackTrace()
            }
        }
    }
}